src/components/user/TargetForm.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { string, bool, func, array } from 'prop-types';
+import { Field, reduxForm } from 'redux-form/immutable';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import {
+  injectIntl,
+  intlShape,
+  defineMessages,
+  FormattedMessage
+} from 'react-intl';
+
+import { getTopics, } from '../../actions/sessionActions';
+import Routes from '../../constants/routesPaths';
+import Loading from '../common/Loading';
+import Input from '../common/Input';
+import SelectBox from '../common/Select';
+import { validations, target } from '../../utils/constraints';
+import smilies from '../../img/smilies.svg';
+import close from '../../img/close.png';
+import targetImg from '../../img/target.svg';
+
+const messages = defineMessages({
+  name: { id: 'signup.name' },
+  firstName: { id: 'edit.firstName' },
+  lastName: { id: 'edit.lastName' },
+  email: { id: 'login.form.email' },
+  topic: { id: 'target.topic' },
+  genderSelect: { id: 'signup.genderSelect' },
+  title: { id: 'target.title' },
+  area: { id: 'target.area' },
+  success: { id: 'edit.success' }
+});
+
+const topicsToJSON = topic => ({
+  value: topic.topic.id,
+  label: topic.topic.label
+});
+
+let TargetForm = ({ error, handleSubmit, submitting, intl, topics, getTopics, eraseTarget }) => {
+  getTopics();
```

Comment:
You should do this in `componentDidMount`, so making this component a class could be good.